dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/blas/MatrixUtils.java

```diff
@@ -0,0 +1,136 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.blas;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Utility class for {@link edu.snu.dolphin.dnn.blas.Matrix}.
+ */
+public final class MatrixUtils {
+
+  private MatrixUtils() {
+  }
+
+  /**
+   * Returns true if and only if two lists of Matrix have same size
+   * and Matrix elements are equal to one another within tolerance.
+   *
+   * @param a one list of Matrix for comparison.
+   * @param b another list of Matrix for comparison.
+   * @param tolerance the maximum difference for which both numbers are still considered equal.
+   * @return true if the two specified lists of Matrix are equal to one another.
+   */
+  public static boolean compare(final List<Matrix> a, final List<Matrix> b, final float tolerance) {
+    if (a.size() != b.size()) {
+      return false;
+    }
+    final Iterator<Matrix> bIter = b.iterator();
+    for (final Matrix m : a) {
+      if (!m.compare(bIter.next(), tolerance)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns true if and only if two arrays of Matrix have same size
+   * and Matrix elements are equal to one another within tolerance.
+   *
+   * @param a one array of Matrix for comparison.
+   * @param b another array of Matrix for comparison.
+   * @param tolerance the maximum difference for which both numbers are still considered equal.
+   * @return true if the two specified arrays of Matrix are equal to one another.
+   */
+  public static boolean compare(final Matrix[] a, final Matrix[] b, final float tolerance) {
+    return compare(Arrays.asList(a), Arrays.asList(b), tolerance);
+  }
+
+
+  /**
+   * Loads a matrix from a Numpy-compatible plain text file with the specified delimiter.
+   * @param matrixFactory a matrix factory used to create a matrix
+   * @param inputStream a Numpy-compatible plain text input stream
+   * @param delimiter a delimiter
+   * @return a loaded matrix
+   * @throws IOException
+   */
+  public static Matrix readNumpy(final MatrixFactory matrixFactory,
+                                 final InputStream inputStream,
+                                 final String delimiter) throws IOException {
+    final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+    String line;
+    final List<float[]> data2 = new ArrayList<>();
```

Comment:
I didn't notice it last time, but the variable name `data2` seems inappropriate. Could we use a better name?